build/update_ext_version.py

```diff
@@ -0,0 +1,102 @@
+import argparse
+import datetime
+import json
+import os
+
+EXT_ROOT = os.path.dirname(os.path.dirname(__file__))
+PACKAGE_JSON_PATH = os.path.join(EXT_ROOT, "package.json")
+
+
+def get_argparse():
+    parser = argparse.ArgumentParser(
+        description="This script updates the python extension micro version based on the release or pre-release channel."
+    )
+    parser.add_argument(
+        "--release",
+        action="store_true",
+        help="Treats the current build is a release build.",
+    )
+    parser.add_argument(
+        "--build-id",
+        action="store",
+        type=int,
+        default=None,
+        help="If persent will be used as a micro version.",
+        required=False,
+    )
+    parser.add_argument(
+        "--for-publishing",
+        action="store_true",
+        help="Preserves `-dev` or `-rc` suffix.",
+    )
+    return parser
+
+
+def is_even(v):
+    return int(v) % 2 == 0
+
+
+def get_build_number():
+    return "1" + datetime.datetime.now(tz=datetime.timezone.utc).strftime("%j%H%M")
+
+
+def parse_version(version):
+    major, minor, parts = version.split(".")
+    try:
+        micro, suffix = parts.split("-")
+    except Exception:
+        micro = parts
+        suffix = ""
+    return (major, minor, micro, suffix)
+
+
+def main():
+    parser = get_argparse()
+    args = parser.parse_args()
+
+    with open(PACKAGE_JSON_PATH, "r", encoding="utf-8") as f:
+        package = json.load(f)
```

Comment:
```suggestion
    package = json.loads(PACKAGE_JSON_PATH.read_text(encoding="utf-8"))
```

Only change if you find it easier to read.